### PR TITLE
Added automated ova deployment module for esxi

### PR DESCRIPTION
**Pull request by nirvishek**
- This is not refactored, need to remove ova-deployment files which are already present in #k8-vmware wrapper (Added those as i refactored multiple classes to individual files).
- Added utility scripts as well (will be relocated to proper locations) for getting vm info, uploading file and executing script inside deployed vm.

### CODE REVIEW

File: scripts/automated-deployment/ova_deployment/ovf_handler.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        try:
+            prog = self.handle.progress()
+            self.lease.Progress(prog)
+            if self.lease.state not in [vim.HttpNfcLease.State.done,
+                                        vim.HttpNfcLease.State.error]:
+                self.start_timer()
+            sys.stderr.write("Progress: %d%%\r" % prog)
+        except:  # Any exception means we should stop updating progress.
+            pass
+
```
Comment (dinis-cruz-gw) [inline scripts/automated-deployment/ova_deployment/ovf_handler.py:140]: actually I think a good amount of this code is already part of the https://github.com/k8-proxy/k8-vmware/pull/33 

can you sync up with @samarthdd to make sure we are not duplicating code

File: scripts/automated-deployment/ova_deployment/tools/cli.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                        required=False,
+                        action='store',
+                        help='Password to use when connecting to host')
+
+    parser.add_argument('-S', '--disable_ssl_verification',
+                        required=False,
+                        action='store_true',
+                        help='Disable ssl host certificate verification')
+
+    return parser
```
Comment (dinis-cruz-gw) [inline scripts/automated-deployment/ova_deployment/tools/cli.py:67]: convert to https://typer.tiangolo.com/

File: scripts/automated-deployment/ova_deployment/tools/tasks.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        if state == vim.TaskInfo.State.success:
+                            # Remove task from taskList
+                            task_list.remove(str(task))
+                        elif state == vim.TaskInfo.State.error:
+                            raise task.info.error
+            # Move to next version
+            version = update.version
+    finally:
+        if pcfilter:
+            pcfilter.Destroy()
```
Comment (dinis-cruz-gw) [inline scripts/automated-deployment/ova_deployment/tools/tasks.py:58]: there is already a class/method in k8-vmware for this

**Review by dinis-cruz-gw — CHANGES_REQUESTED** [review]
please see my nodes and MORE IMPORTANTLY please check the code on k8-vmware before adding it here

a lot of the code below has been refactored into that project. in fact the point of that project is to make sure that scripts/tools like this don't have a lot of code in it 

basically, part of what we want here is to either:
 - leverage the APIs already in k8-vmware APIs
 - improve the k8-vmware APIs
 - create really clean and useful k8-vmware APIs that dramatically reduce the code required to write scripts/tools like this

File: scripts/automated-deployment/ova_deployment/ovf_handler.py (hunk 1 above)
Comment (nirvishek) [reply scripts/automated-deployment/ova_deployment/ovf_handler.py]: @dinis-cruz-gw i have removed this file and now using it from the api itself.

File: scripts/automated-deployment/ova_deployment/VM_deploy_ova.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    try:
+        for rp in containerView.view:
+            if rp.runtime.memory.unreservedForVm > unreservedForVm:
+                largestRp = rp
+                unreservedForVm = rp.runtime.memory.unreservedForVm
+    finally:
+        containerView.Destroy()
+    if largestRp is None:
+        raise Exception("Failed to find a resource pool in dc %s" % dc.name)
+    return largestRp
```
[the remark this answers is not in the document]
Comment (nirvishek) [reply scripts/automated-deployment/ova_deployment/VM_deploy_ova.py]: @dinis-cruz-gw i have refactored this file to class, and the method on the main api is missing one part where i can provide vm name as the argument. will add that to the main API and create PR.

**Pull request merged**